### DISCUSSION TITLE
Ansible Script:  Need to run "invoke superuser" with answers in one pass

### DISCUSSION THREAD

**Discussion by compumatter** [inventree/InvenTree#4066] [Q&A]
Hello,

I am building an Ansible script to install InvenTree in an automated way.   At the end of the install there is a need to run "invoke superuser" and then manually answer the questions about username, email and password

Can these be passed in via some kind of append / flags so the questions are not asked ?

Thanks.

Jay
CompuMatter

**Comment by wolflu05** [comment]
You could try to do something like `echo "answer1\nanswer2" | invoke superuser`. Or inventree needs to add some parameters.

**Reply by wolflu05** [reply]
@compumatter I've tested it and it seems like that this solution doesn't work here. After googling a bit I found this solution. Curious what @matmair has to say to this:
```bash
echo "from django.contrib.auth import get_user_model; User = get_user_model(); User.objects.create_superuser('test', 'admin@myproject.com', 'test')" | python InvenTree/manage.py shell
```

_Originally from https://stackoverflow.com/questions/6244382/how-to-automate-createsuperuser-on-django_

**Reply by compumatter** [reply]
Thanks for the reply and the follow up effort.  Looks like @matmair has a good idea below.

**Reply by wolflu05** [reply]
Yeah, also saw it. I haven't thought in that direction, because I tried to find some command solution. But matmair's solution is even better

**Comment by matmair** [comment]
@compumatter you could set env variables to create the user and just start up a worker: https://docs.inventree.org/en/latest/start/docker_prod/#create-administrator-account

**Reply by compumatter** [reply]
This does indeed look like a solution.  I will go about that and drop back with a report.  Thank you.

**Reply by matmair** [reply]
@compumatter did this work? An Ansible script would be great for the sysadmins in the community. I would be willing to code something up if the suggestion did not work. We have a mechanism via invoke that could easily be used for something like this.

**Reply by compumatter** [reply]
Matmair, just getting back to this.  Will advise and share with the community.

**Reply by compumatter** [reply]
OK, the best solution for me to resolve this was to place the creds it in the config.yaml file just after the files were extracted.  I'm not using Docker for my install. I tried passing it in via environment variables but they seemed to be ignored.  Putting them in config.yaml bypassed the need to run invoke superuser at all.  It can them be pulled out via Ansible when the role runs if you don't want them hanging around.  I've got an Ansible role created.  It works.  That said, I don't have the time right now to make it 'distro flexible'.  It runs under Ubuntu 22.04.  Start to finish it went through each of the install requirements including any custom gunicorn, supervisor configs or pathing needs.  The variables that are currently global would need to be put in defaults.  I'll create a git for that so people can have some code to start with and build upon.
